src/exercises/day-2/points-polygons.rs

```diff
@@ -84,18 +84,12 @@ impl Polygon {
     }
 
     pub fn length(&self) -> f64 {
-        if self.points.is_empty() {
-            return 0.0;
+        let num_points = self.points.len();
+        let mut len: f64 = 0.0;
+        for idx in 0..num_points {
+            len += self.points[idx].dist(self.points[(idx + 1) % num_points]);
```

Comment:
I agree that the code is shorter... However, the code here is a little more _dangerous_ since every indexing operation has the potential to fail (of course it won't fail here because you do the wrap-around).

In Rust, you should normally favor using iterators. That way, the _structure_ of your code ensures that you cannot index outside of the collection.

Reply:
Got it, appreciate you for taking the time to give feedback, sounds logical, agree we should go with your suggestion. Thank  you for pointing this out!